### PR TITLE
GKE: Enable integration tests

### PR DESCRIPTION
**Pull request by sameersbn**
- [x] Configure CI setup for testing on GKE clusters
- [x] Configure DNS suffix for GKE tests
- [x] Clean up disk leakage
- [x] Debug monitoring test failures in GKE

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -335,6 +356,99 @@ az account set -s $AZURE_SUBSCRIPTION_ID
...
+                                            ])
+                                        }
+                                    }
+                                    finally {
+                                        container('gcloud') {
+                                            def disksFilter = "${clusterName}".take(18).replaceAll(/-$/, '')
+                                            sh "gcloud auth activate-service-account --key-file ${GOOGLE_APPLICATION_CREDENTIALS}"
+                                            sh "gcloud container clusters delete ${clusterName} --zone ${zone} --project ${project} --quiet || :"
+                                            sh "gcloud compute disks delete \$(gcloud compute disks list --project ${project} --filter name:${disksFilter} --format='value(name)') --project ${project} --zone ${zone} --quiet || :"
+                                            sh "gcloud dns managed-zones delete \$(gcloud dns managed-zones list --filter dnsName:${dnsZone} --format='value(name)' --project ${project}) --project ${project} || :"
```
Comment (anguslees) [inline Jenkinsfile]: Aside: If it becomes relevant at some point, we can easily delete all the entries in a managed zone using:
```
gcloud dns record-sets import /dev/null --zone=$zone --delete-all-existing
```

File: Jenkinsfile [hunk 2]
```diff
@@ -335,6 +356,99 @@ az account set -s $AZURE_SUBSCRIPTION_ID
...
+                                    try {
+                                        runIntegrationTest(platform, "gke --project=${project} --dns-zone=${dnsZone} --email=${adminEmail} --authz-domain=\"*\"", "") { // --dns-suffix ${dnsZone}
+                                            container('gcloud') {
+                                                sh "gcloud auth activate-service-account --key-file ${GOOGLE_APPLICATION_CREDENTIALS}"
+                                                sh """
+gcloud container clusters create ${clusterName} \
+ --cluster-version ${kversion} \
+ --project ${project} \
+ --create-subnetwork '' \
+ --enable-ip-alias \
```
Comment (anguslees) [inline Jenkinsfile]: I think we should remove `--create-subnetwork "" --enable-ip-alias` ?
I can't remember why I put them there (anticipating one of the many future-compatibility-warnings that `gcloud c c create` reports?), and they aren't mentioned in the gke quickstart.  Better to test whatever we recommend in the docs.

File: Jenkinsfile [hunk 3]
```diff
@@ -224,15 +242,15 @@ spec:
                             // $AZURE_SUBSCRIPTION_ID, $AZURE_TENANT_ID.
                             withCredentials([azureServicePrincipal('jenkins-bkpr-owner-sp')]) {
                                 def resourceGroup = 'jenkins-bkpr-rg'
-                                def clusterName = ("${env.BRANCH_NAME}".take(8) + "-${env.BUILD_NUMBER}-${platform}-" + UUID.randomUUID().toString().take(5)).replaceAll(/[^a-zA-Z0-9-]/, '-').toLowerCase()
+                                def clusterName = ("${env.BRANCH_NAME}".take(8) + "-${env.BUILD_NUMBER}-" + UUID.randomUUID().toString().take(5) + "-${platform}").replaceAll(/[^a-zA-Z0-9-]/, '-').toLowerCase()
```
Comment (anguslees) [inline Jenkinsfile]: We need more uniqueness than platform+BUILD_NUMBER?  This expression is getting a bit silly - I think if we're going to include a random component, then lets just drop all the non-random bits and rely on tags/labels to report meta-info.

File: tests/monitoring_test.go [hunk 4]
```diff
@@ -87,6 +87,11 @@ var _ = Describe("Monitoring", func() {
 	var ns string
 
 	BeforeEach(func() {
+		if *dnsSuffix == "" {
```
[the remark this answers is not in the document]
Comment (sameersbn) [reply tests/monitoring_test.go]: That what I thought, but for some reason the service endpoint is not reachable via the proxy

File: Jenkinsfile (hunk 3 above)
Comment (sameersbn) [reply Jenkinsfile]: The truncation removes the uniqueness. This change essentially is to make sure the uniqueness does not get removed as a result of truncations

File: Jenkinsfile (hunk 2 above)
Comment (sameersbn) [reply Jenkinsfile]: yep I will remove this. It's not needed

File: tests/monitoring_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (sameersbn) [reply tests/monitoring_test.go]: for reference here's one of the instances where the monitoring tests failed without a valid dns setup https://jenkins-bkpr.k.dev.bitnami.net/blue/organizations/jenkins/kube-prod-runtime/detail/private-sbox/35/pipeline/83

[the remark this answers is not in the document]
Comment (sameersbn) [reply tests/monitoring_test.go]: well.. it seems the error is not related to the DNS setup. I got it wrong. Need to find out what is causing the monitoring tests to fail

File: Jenkinsfile (hunk 1 above)
Comment (sameersbn) [reply Jenkinsfile]: thanks for this. Using this now and reverted back to deleting the cluster asynchronously.

File: Jenkinsfile (hunk 2 above)
Comment (sameersbn) [reply Jenkinsfile]: cleaned up

File: tests/monitoring_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (sameersbn) [reply tests/monitoring_test.go]: removed this condition. debugging the reason for monitoring test failures on GKE. Oddly I don't notice any issues while running the tests locally

File: Jenkinsfile [hunk 5]
```diff
@@ -26,15 +29,70 @@ def withGo(Closure body) {
...
+        }
+    }
+}
+
+def insertGlueRecords(String name, String nameServersJson, String ttl, String zone, String resourceGroup) {
+    withCredentials([azureServicePrincipal('jenkins-bkpr-owner-sp')]) {
+        container('az') {
+            sh "az login --service-principal -u \$AZURE_CLIENT_ID -p \$AZURE_CLIENT_SECRET -t \$AZURE_TENANT_ID"
+            sh "az account set -s \$AZURE_SUBSCRIPTION_ID"
+            for (ns in readJSON(text: nameServersJson)) {
```
Comment (anguslees) [inline Jenkinsfile]: Minor: Probably makes sense to make the `nameServers` argument to this function be an array/list of strings and call `readJSON` in the caller(s).  So far 2 out of 2 give it to us as a json-serialised array, so I'm also ok with leaving that as the "native form" for now...

**Pull request merged**